### ISSUE TITLE
Forgot password silently fails

### ISSUE THREAD

**Issue by pdpinch** [mitodl/mitxonline#39]
### Steps to Reproduce

1. Go to /signup/forgot-password
2. Enter the email address of an existing user 

### Expected Behavior

1. Browser updates with a message confirming the email was sent (200 response) 
2. password reset email sent to the email address

### Actual Behavior

1. Browser updates with a message confirming the email was sent 
2. no password resent email is sent (no "Acccepted" message appears in Mailgun logs)
3. POST to https://mitxonline-rc.mitxonline.mit.edu/api/password_reset/ returns a 502 error

Another related issue is that the API call returns an error, but it isn't displayed to the user. We really should let the user know that their attempt to reset their password failed. 

### Stacktrace

Too long to post the whole thing here. I think this is the relevant bit: 

```
2021-07-28T20:45:53.465772+00:00 app[web.1]: Traceback (most recent call last):
2021-07-28T20:45:53.465812+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
2021-07-28T20:45:53.466011+00:00 app[web.1]: response = get_response(request)
2021-07-28T20:45:53.466013+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
2021-07-28T20:45:53.466221+00:00 app[web.1]: response = wrapped_callback(request, *callback_args, **callback_kwargs)
2021-07-28T20:45:53.466226+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/sentry_sdk/integrations/django/views.py", line 67, in sentry_wrapped_callback
2021-07-28T20:45:53.467213+00:00 app[web.1]: return callback(request, *args, **kwargs)
2021-07-28T20:45:53.467264+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_django.py", line 562, in wrapper
2021-07-28T20:45:53.467559+00:00 app[web.1]: return wrapped(*args, **kwargs)
2021-07-28T20:45:53.467586+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
2021-07-28T20:45:53.467730+00:00 app[web.1]: return view_func(*args, **kwargs)
2021-07-28T20:45:53.467731+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/rest_framework/viewsets.py", line 125, in view
2021-07-28T20:45:53.467883+00:00 app[web.1]: return self.dispatch(request, *args, **kwargs)
2021-07-28T20:45:53.467938+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/component_djangorestframework.py", line 60, in _nr_wrapper_APIView_dispatch_
2021-07-28T20:45:53.468102+00:00 app[web.1]: return wrapped(*args, **kwargs)
2021-07-28T20:45:53.468111+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/rest_framework/views.py", line 509, in dispatch
2021-07-28T20:45:53.468398+00:00 app[web.1]: response = self.handle_exception(exc)
2021-07-28T20:45:53.468402+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/component_djangorestframework.py", line 67, in _handle_exception_wrapper
2021-07-28T20:45:53.468564+00:00 app[web.1]: return wrapped(*args, **kwargs)
2021-07-28T20:45:53.468568+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/rest_framework/views.py", line 469, in handle_exception
2021-07-28T20:45:53.468837+00:00 app[web.1]: self.raise_uncaught_exception(exc)
2021-07-28T20:45:53.468841+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
2021-07-28T20:45:53.469118+00:00 app[web.1]: raise exc
2021-07-28T20:45:53.469122+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/rest_framework/views.py", line 506, in dispatch
2021-07-28T20:45:53.469435+00:00 app[web.1]: response = handler(request, *args, **kwargs)
2021-07-28T20:45:53.469436+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/djoser/views.py", line 241, in reset_password
2021-07-28T20:45:53.469624+00:00 app[web.1]: settings.EMAIL.password_reset(self.request, context).send(to)
2021-07-28T20:45:53.469628+00:00 app[web.1]: File "/app/./authentication/views.py", line 160, in send
2021-07-28T20:45:53.469801+00:00 app[web.1]: subject, text_body, html_body = render_email_templates(
2021-07-28T20:45:53.469805+00:00 app[web.1]: File "/app/./mail/api.py", line 138, in render_email_templates
2021-07-28T20:45:53.469958+00:00 app[web.1]: html_text = render_to_string("{}/body.html".format(template_name), context)
2021-07-28T20:45:53.469996+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/loader.py", line 62, in render_to_string
2021-07-28T20:45:53.470134+00:00 app[web.1]: return template.render(context, request)
2021-07-28T20:45:53.470182+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/backends/django.py", line 61, in render
2021-07-28T20:45:53.470338+00:00 app[web.1]: return self.template.render(context)
2021-07-28T20:45:53.470342+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 170, in render
2021-07-28T20:45:53.470533+00:00 app[web.1]: return self._render(context)
2021-07-28T20:45:53.470537+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/api/function_trace.py", line 143, in dynamic_wrapper
2021-07-28T20:45:53.470683+00:00 app[web.1]: return wrapped(*args, **kwargs)
2021-07-28T20:45:53.470735+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 162, in _render
2021-07-28T20:45:53.470935+00:00 app[web.1]: return self.nodelist.render(context)
2021-07-28T20:45:53.470936+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 938, in render
2021-07-28T20:45:53.471412+00:00 app[web.1]: bit = node.render_annotated(context)
2021-07-28T20:45:53.471417+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 905, in render_annotated
2021-07-28T20:45:53.471829+00:00 app[web.1]: return self.render(context)
2021-07-28T20:45:53.471833+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/loader_tags.py", line 150, in render
2021-07-28T20:45:53.472030+00:00 app[web.1]: return compiled_parent._render(context)
2021-07-28T20:45:53.472031+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/api/function_trace.py", line 143, in dynamic_wrapper
2021-07-28T20:45:53.472180+00:00 app[web.1]: return wrapped(*args, **kwargs)
2021-07-28T20:45:53.472181+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 162, in _render
2021-07-28T20:45:53.472409+00:00 app[web.1]: return self.nodelist.render(context)
2021-07-28T20:45:53.472413+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 938, in render
2021-07-28T20:45:53.472828+00:00 app[web.1]: bit = node.render_annotated(context)
2021-07-28T20:45:53.472832+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 905, in render_annotated
2021-07-28T20:45:53.473213+00:00 app[web.1]: return self.render(context)
2021-07-28T20:45:53.473217+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_django.py", line 786, in wrapper
2021-07-28T20:45:53.473518+00:00 app[web.1]: return wrapped(*args, **kwargs)
2021-07-28T20:45:53.473522+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/loader_tags.py", line 62, in render
2021-07-28T20:45:53.473673+00:00 app[web.1]: result = block.nodelist.render(context)
2021-07-28T20:45:53.473674+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 938, in render
2021-07-28T20:45:53.474007+00:00 app[web.1]: bit = node.render_annotated(context)
2021-07-28T20:45:53.474011+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 905, in render_annotated
2021-07-28T20:45:53.474365+00:00 app[web.1]: return self.render(context)
2021-07-28T20:45:53.474369+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/defaulttags.py", line 446, in render
2021-07-28T20:45:53.474587+00:00 app[web.1]: url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
2021-07-28T20:45:53.474592+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_django.py", line 669, in wrapper
2021-07-28T20:45:53.474842+00:00 app[web.1]: return execute(*args, **kwargs)
2021-07-28T20:45:53.474846+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_django.py", line 668, in execute
2021-07-28T20:45:53.475116+00:00 app[web.1]: return wrapped(viewname, *args, **kwargs)
2021-07-28T20:45:53.475158+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/urls/base.py", line 87, in reverse
2021-07-28T20:45:53.475295+00:00 app[web.1]: return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
2021-07-28T20:45:53.475325+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/urls/resolvers.py", line 685, in _reverse_with_prefix
2021-07-28T20:45:53.475599+00:00 app[web.1]: raise NoReverseMatch(msg)
2021-07-28T20:45:53.475653+00:00 app[web.1]: django.urls.exceptions.NoReverseMatch: Reverse for 'password-reset-confirm' with keyword arguments '{'uid': 'Mg', 'token': 'aqikcf-8da67f5b20ccb244643871bc6a88b583'}' not found. 1 pattern(s) tried: ['signin/forgot-password/confirm/(?P<uid>[0-9A-Za-z_\\-]+)/(?P<token>[0-9A-Za-z]{1,13}-[0-9A-Za-z]{1,20})/$']
2021-07-28T20:45:53.475671+00:00 app[web.1]: 
2021-07-28T20:45:53.475672+00:00 app[web.1]: During handling of the above exception, another exception occurred:
2021-07-28T20:45:53.475672+00:00 app[web.1]: 
2021-07-28T20:45:53.475703+00:00 app[web.1]: Traceback (most recent call last):
2021-07-28T20:45:53.475763+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
2021-07-28T20:45:53.475899+00:00 app[web.1]: response = get_response(request)
2021-07-28T20:45:53.475920+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/sentry_sdk/integrations/django/middleware.py", line 175, in __call__
2021-07-28T20:45:53.476748+00:00 app[web.1]: return f(*args, **kwargs)
2021-07-28T20:45:53.476796+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django_user_agents/middleware.py", line 14, in __call__
2021-07-28T20:45:53.476933+00:00 app[web.1]: return self.get_response(request)
2021-07-28T20:45:53.476937+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_django.py", line 1205, in _wrapper
2021-07-28T20:45:53.477379+00:00 app[web.1]: return wrapped(*args, **kwargs)
2021-07-28T20:45:53.477423+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/core/handlers/exception.py", line 49, in inner
2021-07-28T20:45:53.477593+00:00 app[web.1]: response = response_for_exception(request, exc)
2021-07-28T20:45:53.477633+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/core/handlers/exception.py", line 103, in response_for_exception
2021-07-28T20:45:53.477779+00:00 app[web.1]: response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
2021-07-28T20:45:53.477831+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_django.py", line 508, in wrapper
2021-07-28T20:45:53.478108+00:00 app[web.1]: return _wrapped(*args, **kwargs)
2021-07-28T20:45:53.478144+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_django.py", line 501, in _wrapped
2021-07-28T20:45:53.478426+00:00 app[web.1]: return wrapped(request, resolver, exc_info)
2021-07-28T20:45:53.478455+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/core/handlers/exception.py", line 142, in handle_uncaught_exception
2021-07-28T20:45:53.478607+00:00 app[web.1]: return callback(request, **param_dict)
2021-07-28T20:45:53.478632+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_django.py", line 562, in wrapper
2021-07-28T20:45:53.478992+00:00 app[web.1]: return wrapped(*args, **kwargs)
2021-07-28T20:45:53.478997+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/utils/decorators.py", line 130, in _wrapped_view
2021-07-28T20:45:53.479248+00:00 app[web.1]: response = view_func(request, *args, **kwargs)
2021-07-28T20:45:53.479298+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/views/defaults.py", line 97, in server_error
2021-07-28T20:45:53.479485+00:00 app[web.1]: return HttpResponseServerError(template.render())
2021-07-28T20:45:53.479496+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/backends/django.py", line 61, in render
2021-07-28T20:45:53.479665+00:00 app[web.1]: return self.template.render(context)
2021-07-28T20:45:53.479700+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 170, in render
2021-07-28T20:45:53.479894+00:00 app[web.1]: return self._render(context)
2021-07-28T20:45:53.479945+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/api/function_trace.py", line 143, in dynamic_wrapper
2021-07-28T20:45:53.480120+00:00 app[web.1]: return wrapped(*args, **kwargs)
2021-07-28T20:45:53.480170+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 162, in _render
2021-07-28T20:45:53.480421+00:00 app[web.1]: return self.nodelist.render(context)
2021-07-28T20:45:53.480426+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 938, in render
2021-07-28T20:45:53.480849+00:00 app[web.1]: bit = node.render_annotated(context)
2021-07-28T20:45:53.480868+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 905, in render_annotated
2021-07-28T20:45:53.481335+00:00 app[web.1]: return self.render(context)
2021-07-28T20:45:53.481340+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/loader_tags.py", line 150, in render
2021-07-28T20:45:53.481472+00:00 app[web.1]: return compiled_parent._render(context)
2021-07-28T20:45:53.481499+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/api/function_trace.py", line 143, in dynamic_wrapper
2021-07-28T20:45:53.481660+00:00 app[web.1]: return wrapped(*args, **kwargs)
2021-07-28T20:45:53.481661+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 162, in _render
2021-07-28T20:45:53.481863+00:00 app[web.1]: return self.nodelist.render(context)
2021-07-28T20:45:53.481868+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 938, in render
2021-07-28T20:45:53.482278+00:00 app[web.1]: bit = node.render_annotated(context)
2021-07-28T20:45:53.482309+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 905, in render_annotated
2021-07-28T20:45:53.482664+00:00 app[web.1]: return self.render(context)
2021-07-28T20:45:53.482670+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/defaulttags.py", line 401, in render
2021-07-28T20:45:53.482916+00:00 app[web.1]: return strip_spaces_between_tags(self.nodelist.render(context).strip())
2021-07-28T20:45:53.482921+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 938, in render
2021-07-28T20:45:53.483331+00:00 app[web.1]: bit = node.render_annotated(context)
2021-07-28T20:45:53.483336+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 905, in render_annotated
2021-07-28T20:45:53.483838+00:00 app[web.1]: return self.render(context)
2021-07-28T20:45:53.483875+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/library.py", line 192, in render
2021-07-28T20:45:53.484039+00:00 app[web.1]: output = self.func(*resolved_args, **resolved_kwargs)
2021-07-28T20:45:53.484067+00:00 app[web.1]: File "/app/./main/templatetags/js_interop.py", line 15, in js_settings
2021-07-28T20:45:53.484267+00:00 app[web.1]: request = context["request"]
2021-07-28T20:45:53.484268+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/context.py", line 83, in __getitem__
2021-07-28T20:45:53.484352+00:00 app[web.1]: raise KeyError(key)
2021-07-28T20:45:53.484368+00:00 app[web.1]: KeyError: 'request'
2021-07-28T20:45:53.484401+00:00 app[web.1]: 
2021-07-28T20:45:53.484402+00:00 app[web.1]: During handling of the above exception, another exception occurred:
2021-07-28T20:45:53.484402+00:00 app[web.1]: 
2021-07-28T20:45:53.484412+00:00 app[web.1]: Traceback (most recent call last):
2021-07-28T20:45:53.484447+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
2021-07-28T20:45:53.484574+00:00 app[web.1]: response = get_response(request)
2021-07-28T20:45:53.484605+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/sentry_sdk/integrations/django/middleware.py", line 175, in __call__
2021-07-28T20:45:53.484795+00:00 app[web.1]: return f(*args, **kwargs)
2021-07-28T20:45:53.484839+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/utils/deprecation.py", line 114, in __call__
2021-07-28T20:45:53.485048+00:00 app[web.1]: response = response or self.get_response(request)
2021-07-28T20:45:53.485081+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_django.py", line 1205, in _wrapper
2021-07-28T20:45:53.485578+00:00 app[web.1]: return wrapped(*args, **kwargs)
2021-07-28T20:45:53.485583+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/core/handlers/exception.py", line 49, in inner
2021-07-28T20:45:53.485722+00:00 app[web.1]: response = response_for_exception(request, exc)
2021-07-28T20:45:53.485727+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/core/handlers/exception.py", line 103, in response_for_exception
2021-07-28T20:45:53.485853+00:00 app[web.1]: response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
2021-07-28T20:45:53.485894+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_django.py", line 508, in wrapper
2021-07-28T20:45:53.486217+00:00 app[web.1]: return _wrapped(*args, **kwargs)
2021-07-28T20:45:53.486250+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_django.py", line 501, in _wrapped
2021-07-28T20:45:53.486481+00:00 app[web.1]: return wrapped(request, resolver, exc_info)
2021-07-28T20:45:53.486515+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/core/handlers/exception.py", line 142, in handle_uncaught_exception
2021-07-28T20:45:53.486653+00:00 app[web.1]: return callback(request, **param_dict)
2021-07-28T20:45:53.486657+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/hooks/framework_django.py", line 562, in wrapper
2021-07-28T20:45:53.486890+00:00 app[web.1]: return wrapped(*args, **kwargs)
2021-07-28T20:45:53.486894+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/utils/decorators.py", line 130, in _wrapped_view
2021-07-28T20:45:53.487033+00:00 app[web.1]: response = view_func(request, *args, **kwargs)
2021-07-28T20:45:53.487066+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/views/defaults.py", line 97, in server_error
2021-07-28T20:45:53.487236+00:00 app[web.1]: return HttpResponseServerError(template.render())
2021-07-28T20:45:53.487266+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/backends/django.py", line 61, in render
2021-07-28T20:45:53.487398+00:00 app[web.1]: return self.template.render(context)
2021-07-28T20:45:53.487440+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 170, in render
2021-07-28T20:45:53.487603+00:00 app[web.1]: return self._render(context)
2021-07-28T20:45:53.487638+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/api/function_trace.py", line 143, in dynamic_wrapper
2021-07-28T20:45:53.487783+00:00 app[web.1]: return wrapped(*args, **kwargs)
2021-07-28T20:45:53.487812+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 162, in _render
2021-07-28T20:45:53.487980+00:00 app[web.1]: return self.nodelist.render(context)
2021-07-28T20:45:53.487984+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 938, in render
2021-07-28T20:45:53.488353+00:00 app[web.1]: bit = node.render_annotated(context)
2021-07-28T20:45:53.488437+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 905, in render_annotated
2021-07-28T20:45:53.488712+00:00 app[web.1]: return self.render(context)
2021-07-28T20:45:53.488740+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/loader_tags.py", line 150, in render
2021-07-28T20:45:53.488882+00:00 app[web.1]: return compiled_parent._render(context)
2021-07-28T20:45:53.488909+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/newrelic/api/function_trace.py", line 143, in dynamic_wrapper
2021-07-28T20:45:53.489060+00:00 app[web.1]: return wrapped(*args, **kwargs)
2021-07-28T20:45:53.489086+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 162, in _render
2021-07-28T20:45:53.489254+00:00 app[web.1]: return self.nodelist.render(context)
2021-07-28T20:45:53.489282+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 938, in render
2021-07-28T20:45:53.489629+00:00 app[web.1]: bit = node.render_annotated(context)
2021-07-28T20:45:53.489634+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 905, in render_annotated
2021-07-28T20:45:53.489967+00:00 app[web.1]: return self.render(context)
2021-07-28T20:45:53.489972+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/defaulttags.py", line 401, in render
2021-07-28T20:45:53.490234+00:00 app[web.1]: return strip_spaces_between_tags(self.nodelist.render(context).strip())
2021-07-28T20:45:53.490235+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 938, in render
2021-07-28T20:45:53.490506+00:00 app[web.1]: bit = node.render_annotated(context)
2021-07-28T20:45:53.490538+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/base.py", line 905, in render_annotated
2021-07-28T20:45:53.490861+00:00 app[web.1]: return self.render(context)
2021-07-28T20:45:53.490890+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/library.py", line 192, in render
2021-07-28T20:45:53.491045+00:00 app[web.1]: output = self.func(*resolved_args, **resolved_kwargs)
2021-07-28T20:45:53.491050+00:00 app[web.1]: File "/app/./main/templatetags/js_interop.py", line 15, in js_settings
2021-07-28T20:45:53.491214+00:00 app[web.1]: request = context["request"]
2021-07-28T20:45:53.491219+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.9/site-packages/django/template/context.py", line 83, in __getitem__
2021-07-28T20:45:53.491348+00:00 app[web.1]: raise KeyError(key)
2021-07-28T20:45:53.491376+00:00 app[web.1]: KeyError: 'request'
2021-07-28T20:45:53.491380+00:00 app[web.1]: 
2021-07-28T20:45:53.491380+00:00 app[web.1]: During handling of the above exception, another exception occurred:
2021-07-28T20:45:53.491381+00:00 app[web.1]: 
2021-07-28T20:45:53.491409+00:00 app[web.1]: Traceback (most recent call last):
```

**Comment by arslanashraf7** [comment]
For the sake of posterity:

**What caused template reverse error**
While debugging this issue I figured out that the template rendering issue was actually related to the regex mismatch of the `token` parameter in our `forgot-password` [API](https://github.com/mitodl/mitxonline/blob/main/main/urls.py#L49).

**What caused regex mismatch and why we didn't face it in other apps**
The `regex` mismatch was caused due to the version of `Django`.

The package(`Djoser`) we used for resetting passwords actually [generates a token](https://github.com/sunscrapers/djoser/blob/master/djoser/email.py#L35) while [sending the email](https://github.com/mitodl/mitxonline/blob/main/authentication/views.py#L151) to authenticate the reset password link. This project is using `Django==3.1.12` and the token length differs between Django versions 2 and 3 e.g. (in Django 2 the generated token length used to be `24` but in Django3 it is `39`) which failed out regex.
Django's token generation reference is [here](https://github.com/django/django/blob/f2ed2211c26ba375390cb76725c95ae970a0fd1d/django/contrib/auth/tokens.py#L62).

**Comment by pdpinch** [comment]
closed via #44
